src/github-app-setup.js

```diff
@@ -0,0 +1,68 @@
+const {
+  error,
+  getInput,
+  info,
+  setFailed,
+  setSecret
+} = require('@actions/core')
+const { context, getOctokit } = require('@actions/github')
+const { App } = require('@octokit/app')
+const isBase64 = require('is-base64')
+const { execSync } = require('child_process')
+const { convertActionToCloneCommand } = require('./action-parser')
+
+// Code based on https://github.com/tibdex/github-app-token
+async function obtainAppToken () {
+  try {
+    // Get the parameters and throw if they are not found
+    const id = Number(getInput('app_id', { required: true }))
```

Comment:
Maybe if we have the appId as an input param here (`obtainAppToken`) we could avoid having to call `getInput('app_id')` more than once.

Reply:
Agreed. Although this doesn't have tests now, it would make that code certainly more testable. Will change it